### PR TITLE
[AIE] Memory bank modelling in the frontend/backend

### PR DESCRIPTION
**Pull request by krishnamtibrewala**
Reopening #93 for review comments kindly refer #93

### CODE REVIEW

**Review by andcarminati — APPROVED** [review]
LGTM, ready to land.

File: clang/test/CodeGen/aie/aie2/aie2-bank-annotation.cpp [hunk 1]
```diff
@@ -0,0 +1,145 @@
+// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py UTC_ARGS: --version 4
```
Comment (gbossu) [inline clang/test/CodeGen/aie/aie2/aie2-bank-annotation.cpp:1]: Is this really auto-generated? There are lines at the bottom that look manual:
```
//.
// CHECK: [[TBAA2]] = !{[[META3:![0-9]+]], [[META3]], i64 0}
// CHECK: [[META3]] = !{!"int", [[META4:![0-9]+]], i64 0}
// CHECK: [[META4]] = !{!"omnipotent char", [[META5:![0-9]+]], i64 0}
// CHECK: [[META5]] = !{!"Simple C++ TBAA"}
//.
```

**Review by gbossu — APPROVED** [review]
LGTM, great job and thanks for addressing the comments from https://github.com/Xilinx/llvm-aie/pull/93

File: clang/test/CodeGen/aie/aie2/aie2-bank-annotation.cpp (hunk 1 above)
Comment (krishnamtibrewala) [reply clang/test/CodeGen/aie/aie2/aie2-bank-annotation.cpp:1]: These were auto generated.

**Pull request merged**